frame/bounties/src/lib.rs

```diff
@@ -653,12 +780,596 @@ decl_module! {
+							let balance = T::Currency::free_balance(&bounty_account);
+							// reserve the fund for master curator fee
+							bounty.subbounty_free_balance = balance.saturating_sub(bounty.fee);
+						}
+
+						let tst_new_free_balance = bounty.subbounty_free_balance.saturating_sub(value);
+
+						ensure!(tst_new_free_balance > Zero::zero(), Error::<T>::InsufficientBountyBalance);
+
+						// Increment the subbounty tracking count
+						// And i
+						bounty.subbounty_count += 1;
+						match bounty.active_subbounty.binary_search_by_key(
+							&bounty.subbounty_count,
+							|x| *x,
+						) {
+							Ok(_) => {
+								//This should not occur
+							},
+							Err(i) => bounty.active_subbounty.insert(i, bounty.subbounty_count),
+						}
+
+						let _ = Self::create_subbounty(
+							bounty_id,
+							bounty.subbounty_count,
+							description,
+							value
+						);
+
+						// subbounty addition OK, update the subbounty free balance.
+						bounty.subbounty_free_balance = tst_new_free_balance;
+					},
+					_ => return Err(Error::<T>::UnexpectedStatus.into()),
+				}
+				Ok(())
+			})?;
+		}
+
+		/// Propose subcurator for funded subbounty.
+		///
+		/// The dispatch origin for this call must be master curator.
+		///
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work.
+		///
+		/// Proposed subcurator may be "master curator",
+		/// and subcurator fee & curator deposit
+		/// may be Zero in this case.
+		///
+		/// Subbounty must be in "Funded" state, for
+		/// processing the call. and state of subbounty is
+		/// moved to CuratorProposed on successful call
+		/// completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		/// - `subcurator`: Address of subcurator.
+		/// - `fee`: payment fee to subcurator for execution.
+		#[weight = 10_000]
+		fn propose_subcurator(origin, #[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+			subcurator: <T::Lookup as StaticLookup>::Source,
+			#[compact] fee: BalanceOf<T>,
+		) {
+			let signer = ensure_signed(origin)?;
+			let subcurator = T::Lookup::lookup(subcurator)?;
+
+			// Ensure parent bounty is Active & get status of curator
+			let ( master_curator, _) = Self::ensure_bounty_active(bounty_id)?;
+
+			// Mutate the Subbounty instance
+			SubBounties::<T>::try_mutate_exists(bounty_id,
+				subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+
+				let mut subbounty = maybe_subbounty
+					.as_mut()
+					.ok_or(Error::<T>::InvalidIndex)?;
+
+				// Ensure sure caller is curator
+				ensure!( signer == master_curator, Error::<T>::RequireCurator);
+
+				// Ensure subbounty is in expected state
+				match subbounty.status {
+					SubBountyStatus::Added => {},
+					_ => return Err(Error::<T>::UnexpectedStatus.into()),
+				};
+
+				// if curator self assign as subcurator, fee is 0
+				if master_curator != subcurator {
```

Comment:
not sure if we want this. it makes thing more complicated.

Reply:
Do we need to consider, "curator deposit" even if master curator self assign as subcurator ?